website/docs/configuration/provider-requirements.html.md

```diff
@@ -38,66 +82,45 @@ terraform {
 }
 ```
 
-The `required_providers` block must be nested inside a
-[`terraform` block](terraform.html). The `terraform` block can include other
-settings too, but we'll only focus on `required_providers` here.
+The `required_providers` block must be nested inside the top-level
+[`terraform` block](terraform.html) (which can also contain other settings).
 
-The keys inside the `required_providers` block represent each provider's
-[local name](#local-names), which is the unique identifier for a provider within
-a particular module. Each item inside the `required_providers` block is an
-object expecting the following arguments:
+Each argument in the `required_providers` block enables one provider. The key
+determines the provider's [local name](#local-names) (its unique identifier
+within this module), and the value is an object with the following elements:
 
 * `source` - the global [source address](#source-addresses) for the
   provider you intend to use, such as `hashicorp/aws`.
 
 * `version` - a [version constraint](#version-constraints) specifying
   which subset of available provider versions the module is compatible with.
 
--> **Note:** The `required_providers` object syntax described above was added in Terraform v0.13. Previous versions of Terraform used a single string instead of an object, with the string specifying only a version constraint. For example, `mycloud = "~> 1.0"`. Explicit provider source addresses are supported only in Terraform v0.13 and later. If you want to write a module that works with both Terraform v0.12 and v0.13, see [v0.12-Compatible Provider Requirements](#v012-compatible-provider-requirements) below.
+-> **Note:** The `name = { source, version }` syntax for `required_providers`
+was added in Terraform v0.13. Previous versions of Terraform used a version
+constraint string instead of an object (like `mycloud = "~> 1.0"`), and had no
+way to specify provider source addresses. If you want to write a module that
+works with both Terraform v0.12 and v0.13, see [v0.12-Compatible Provider
+Requirements](#v012-compatible-provider-requirements) below.
 
-### Source Addresses
-
-A provider _source address_ both globally identifies a particular provider and
-specifies the primary location from which Terraform can download it.
-Source addresses consist of three parts delimited by slashes (`/`), as
-follows:
+## Names and Addresses
 
-* **Hostname**: the hostname of the Terraform registry that indexes the provider.
-  You can omit the hostname portion and its following slash if the provider
-  is hosted on [the public Terraform Registry](https://registry.terraform.io/),
-  whose hostname is `registry.terraform.io`.
+Each provider has two names:
```

Comment:
It seems weird to refer to the source as a second name. I assume this is a thing that we don't have any say in because it happens other places?

Reply:
Is it weird? I thought it seemed intuitive -- there are lots of things we deal with in real life that have a long name and a short name. Like... me, for instance. 

The source address does have a second function, of showing where to get the provider. But sometimes that's a lie, because you can use an arbitrary placeholder hostname if you're distributing stuff locally. Whereas, it _always_ functions as the provider's globally unique identifier.

Reply:
I guess it just doesn't make sense to me to introduce another term in addition to source. This seems like a device to get around the fact that the source is sometimes arbitrary, but you still have to explain that; drawing an analogy with a name makes the explanation longer rather than shorter. 

If a user told you "I don't know what to set for the provider name" would you assume they might be talking about `source`? If so I think it's ok to leave, but if you mind always jumps to the short name then I think calling the source a name makes things muddier than they need to be.

Reply:
Well, it doesn't really extend beyond this section anyway. I'll switch it to something like "two identifiers - name and address".